Dockerfile

```
@@ -11,7 +11,10 @@ RUN go mod download
 COPY . .
 
 FROM base AS builder
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags='-s -w -extldflags "-static"' .
+
+ARG TARGETPLATFORM
+
```

```suggestion
ENV GOARCH=$TARGETARCH
```

afaict you will otherwise build `amd64` all the time